package/MDAnalysis/converters/OpenMMParser.py

```diff
@@ -123,24 +147,62 @@ def _mda_topology_from_omm_topology(self, omm_topology):
         bond_orders = [b.order for b in omm_topology.bonds()]
         bond_types = [b.type for b in omm_topology.bonds()]
 
-        n_atoms = len(atomids)
-        n_residues = len(resids)
-        n_segments = len(segids)
-
         attrs = [
             Atomids(np.array(atomids, dtype=np.int32)),
             Atomnames(np.array(atomnames, dtype=object)),
-            Atomtypes(np.array(atomtypes, dtype=object)),
             Bonds(bonds, types=bond_types, order=bond_orders, guessed=False),
             ChainIDs(np.array(chainids, dtype=object)),
-            Elements(np.array(elements, dtype=object)),
-            Masses(np.array(masses, dtype=np.float32)),
             Resids(resids),
             Resnums(resnums),
             Resnames(resnames),
             Segids(segids),
         ]
 
+        validated_elements = []
+        masses = []
+        atomtypes = []
+        warn = False
+        for a in omm_topology.atoms():
+            elem = a.element
+            if elem is not None:
+                if elem.symbol.capitalize() in SYMB2Z:
+                    validated_elements.append(elem.symbol)
+                else:
+                    validated_elements.append('')
+                atomtypes.append(elem.symbol)
+                masses.append(elem.mass.value_in_unit(daltons))
+            else:
+                warn = True
+                validated_elements.append('')
+                masses.append(0.0)
+                atomtypes.append('X')
+
+        if warn:
```

Comment:
I'm trying to follow the logic here, maybe I'm missing something obvious but doesn't this end up in some cases giving two conflicting warnings, and in others no warnings at all when there should be one?

So in the case of two conflicting warnings, technically you will set `warn` to `True` if all the elements are None right? So this warning here is triggered, then the warning below? So now users get told "hey your element information is going to be available as empty records", and then straight afterwards "hey I won't populate your element information".

In the case of no warning at all, you could hit line 171 as many times as you want and never generate a user warning. Users never know they are missing elements and carry on eventually getting downstream errors whenever they try to use something that needs elements.

Rather than using and setting a `warn` variable how about the following pseudo-code?
Note: it's not pretty, and calls to any and all aren't free, but hopefully they don't cost too much? (would need checking)

```python
if not all(validated_elements):
    if any(validated_elements):
        warn about missing element info for some atoms
        append elements
        if any([i=='X' for i in atomtypes]):
            warn about atomtypes and masses
    else:
        guess types & masses and warn about lack of elements

Reply:
Thanks for pointing that out, it was a potential issue. I timed the previous commit and the current one using asv and apparently using any/all  doesn't cost too much.